diff --git a/lib/slack/web/api/endpoints/views.rb b/lib/slack/web/api/endpoints/views.rb
index cfa83a5..038c5f0 100644
--- a/lib/slack/web/api/endpoints/views.rb
+++ b/lib/slack/web/api/endpoints/views.rb
@@ -19,6 +19,12 @@ module Slack
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/views/views.open.json
           def views_open(options = {})
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            raise ArgumentError, 'One of :trigger_id, :interactivity_pointer is required' unless options[:trigger_id].nil? ^ options[:interactivity_pointer].nil?
+            if options.key?(:view)
+              view = options[:view]
+              view = JSON.dump(view) unless view.is_a?(String)
+              options = options.merge(view: view)
+            end
             post('views.open', options)
           end
 
@@ -36,6 +42,11 @@ module Slack
           def views_publish(options = {})
             raise ArgumentError, 'Required arguments :user_id missing' if options[:user_id].nil?
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            if options.key?(:view)
+              view = options[:view]
+              view = JSON.dump(view) unless view.is_a?(String)
+              options = options.merge(view: view)
+            end
             post('views.publish', options)
           end
 
@@ -52,6 +63,12 @@ module Slack
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/views/views.push.json
           def views_push(options = {})
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            raise ArgumentError, 'One of :trigger_id, :interactivity_pointer is required' unless options[:trigger_id].nil? ^ options[:interactivity_pointer].nil?
+            if options.key?(:view)
+              view = options[:view]
+              view = JSON.dump(view) unless view.is_a?(String)
+              options = options.merge(view: view)
+            end
             post('views.push', options)
           end
 
@@ -70,6 +87,12 @@ module Slack
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/views/views.update.json
           def views_update(options = {})
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            raise ArgumentError, 'One of :external_id, :view_id is required' unless options[:external_id].nil? ^ options[:view_id].nil?
+            if options.key?(:view)
+              view = options[:view]
+              view = JSON.dump(view) unless view.is_a?(String)
+              options = options.merge(view: view)
+            end
             post('views.update', options)
           end
         end
