diff --git a/lib/slack/web/api/endpoints/views.rb b/lib/slack/web/api/endpoints/views.rb
index 2102b07..6b2f97c 100644
--- a/lib/slack/web/api/endpoints/views.rb
+++ b/lib/slack/web/api/endpoints/views.rb
@@ -19,6 +19,7 @@ module Slack
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/views/views.open.json
           def views_open(options = {})
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            raise ArgumentError, 'One of :trigger_id, :interactivity_pointer is required' unless options[:trigger_id].nil? ^ options[:interactivity_pointer].nil?
             options = encode_options_as_json(options, %i[view])
             post('views.open', options)
           end
@@ -54,6 +55,7 @@ module Slack
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/views/views.push.json
           def views_push(options = {})
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            raise ArgumentError, 'One of :trigger_id, :interactivity_pointer is required' unless options[:trigger_id].nil? ^ options[:interactivity_pointer].nil?
             options = encode_options_as_json(options, %i[view])
             post('views.push', options)
           end
@@ -73,6 +75,7 @@ module Slack
           # @see https://github.com/slack-ruby/slack-api-ref/blob/master/methods/views/views.update.json
           def views_update(options = {})
             raise ArgumentError, 'Required arguments :view missing' if options[:view].nil?
+            raise ArgumentError, 'One of :external_id, :view_id is required' unless options[:external_id].nil? ^ options[:view_id].nil?
             options = encode_options_as_json(options, %i[view])
             post('views.update', options)
           end
